django/db/models/fields/related.py

```diff
@@ -1586,7 +1586,15 @@ class ForeignObject(RelatedField):
 
     def __init__(self, to, from_fields, to_fields, rel=None, related_name=None,
             related_query_name=None, limit_choices_to=None, parent_link=False,
-            on_delete=CASCADE, swappable=True, **kwargs):
+            on_delete=None, swappable=True, **kwargs):
+
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required kwarg for %s in Django 2.0. "
+                "Set the attribute explicitly instead." % self.__class__.__name__,
```

Comment:
This is a nitpick, but `on_delete` is not an attribute you set, it is an argument you pass. I think that second sentence could just be removed. What might be more useful here is a stable link to the `on_delete` docs.